pytest_reana/test_utils.py

```diff
@@ -13,10 +13,17 @@
 
 
 def make_mock_api_client(component):
-    mock_http_client, mock_result, mock_response = Mock(), Mock(), Mock()
+
+    mock_response = Mock()
     mock_response.status_code = 200
-    mock_result.result.return_value = ('_', mock_response)
-    mock_http_client.request.return_value = mock_result
-    mock_api_client = BaseAPIClient(component,
-                                    http_client=mock_http_client)
-    return mock_api_client._client
+    mock_response.raw_bytes = b'Sample downloaded data'
+
+    def mock_api_client(mock_response=mock_response):
```

Comment:
Shall we just pass the expected `status_code` and the expected `result` (documenting of course that one is an integer representing and HTTP status code, and the other a binary string with the expected return value) and then hide inside the function the complexity of doing: `mock_response = Mock()`, `mock_response.status_code = status_code `, `mock_response.raw_bytes = result`. Otherwise we are forcing the clients to do that on their own... and in a test you care about the HTTP status, the result and something else might come in the future which we can add but not creating more mocks.

Also, I think we should add it as fixture since there is no much different between this and the [`consume_queue`](https://github.com/reanahub/pytest-reana/blob/master/pytest_reana/fixtures.py#L311) fixture.

Reply:
Then, we still can do it, we just pass the expected content value and we provide the needed attributes and functions inside...

```python
mock_response.json = lambda expected_content: json.dumps(expected_content)
mock_response.raw_bytes = bytes(expected_content.encode('utf-8'))
```
I don't think we should populate all tests with the same `Mock` initialization everywhere when it can be easily centralised here.

Reply:
Sure there is a way to do it, just this would have to change when there is a test case expecting something different. For now I would leave it free since either way we save 1 line with the `Mock()` instantiation, and when most cases are covered move it here.